src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,68 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String COMA = ",";
+    private static final String SUPPLY = "supply";
+    private static final String BUY = "buy";
+    private static final String RESULT = "result";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        String read = readFromFile(fromFileName);
+        String report = createReport(read);
+        writeToFile(toFileName, report);
+    }
+
+    public String readFromFile(String fromFileName) {
+        File fromFile = new File(fromFileName);
+        StringBuilder stringBuilder = new StringBuilder();
+        try {
+            BufferedReader bufferedReader = new BufferedReader(new FileReader(fromFile));
+            String line = bufferedReader.readLine();
+            do {
+                stringBuilder.append(line).append(COMA);
+                line = bufferedReader.readLine();
+            } while (line != null);
+            bufferedReader.close();
```

Comment:
**try-with-resources**
Remember, if you are using classes that implement an AutoCloseable interface, we should use it with try-with-resources.
https://mate-academy.github.io/jv-program-common-mistakes/java-core/builder-file/work-with-file

Reply:
+